addons/payment_stripe/models/payment.py

```diff
@@ -74,8 +74,10 @@ def _add_available_payment_method_types(self, stripe_session_data, tx_values):
             PMT('p24', ['pl'], ['eur', 'pln'], 'punctual'),
         ]
 
+        enabled = [pmt.name.lower() for pmt in self.payment_icon_ids]
+        enabled_filtered = filter(lambda pmt: pmt.name == 'card' or pmt.name in enabled, all_payment_method_types)
```

Comment:
```suggestion
        icon_names = [icon.name.lower() for icon in self.payment_icon_ids]
        icon_filtered = filter(lambda pmt: pmt.name == 'card' or pmt.name in icon_names, all_payment_method_types)
```
Naming `payment.icon` records "pmt" or "enabled" is confusing. Both concepts are very similar yet a tad different so better not mixing them.